cdap-gateway/src/test/java/co/cask/cdap/gateway/router/NettyRouterTestBase.java

```diff
@@ -388,6 +364,29 @@ public void testConnectionClose() throws Exception {
     Assert.assertEquals(times, defaultServer1.getNumRequests() + defaultServer2.getNumRequests());
   }
 
+  @Test
+  public void testConnectionNoIdleTimeout() throws Exception {
+    // 4700ms is just under the configured idle timeout of 5 seconds
+    URL url = new URL(resolveURI(Constants.Router.GATEWAY_DISCOVERY_NAME, "/v1/timeout/4700"));
```

Comment:
If you have access to `cConf` over is here, then I think it is a good idea to subtract a few hundred millis from `Constants.Router.CONNECTION_TIMEOUT_SECS` rather than hard-coding.